### ISSUE TITLE
Feature request: filter input dataset by other ids

### ISSUE THREAD

**Issue by seanhatton** [nipreps/mriqc#760]
Hi all,

Would it be possible to include additional filters for processing and report generation, such as:

--acq-id  filter input dataset by acquisition id
--ce-id  filter input dataset by contrast enhanced id
--mod-id filter input dataset by modification id

Sean Hatton
University of California, San Diego

**Comment by naveau** [comment]
I would be also interested in this feature. I tried to dig a little bit in the code and found different places where PR could be made :

- Adding new options for filtering at the command line call here :
https://github.com/poldracklab/mriqc/blob/f6c7066fd0dcf7cf41ac77cad1bf3c5254f3bb6d/mriqc/bin/mriqc_run.py#L65-L74

- Then in the main function, options are forwarded to `init_mrqc`
https://github.com/poldracklab/mriqc/blob/f6c7066fd0dcf7cf41ac77cad1bf3c5254f3bb6d/mriqc/bin/mriqc_run.py#L213

- In `init_mrqc` options should be pass explicitly to `collect_bids_data`
https://github.com/poldracklab/mriqc/blob/f6c7066fd0dcf7cf41ac77cad1bf3c5254f3bb6d/mriqc/bin/mriqc_run.py#L401-L408

- In `collect_bids_data` we could then add options to the BIDSLayout filtering API
https://github.com/poldracklab/mriqc/blob/f6c7066fd0dcf7cf41ac77cad1bf3c5254f3bb6d/mriqc/utils/bids.py#L19-L24

Does it make sense ?

**Comment by RebeccaAHayes** [comment]
Our lab could really use this functionality, as we are working on lowfield scans that often cause MRIQC to fail.  Being able to filter with the acq- tag would be ideal.

**Comment by effigies** [comment]
The most flexible option is, as @Remi-Gau says, to add `--bids-filter-file`. @naveau identified the correct places to update.

Original PRs to implement this in fMRIPrep:

* https://github.com/nipreps/niworkflows/pull/407
* https://github.com/nipreps/fmriprep/pull/1770

Current code:

* `collect_data()`: https://github.com/nipreps/niworkflows/blob/26bd24f8d03bdf3e55fe108f38fd1bc505eb7188/niworkflows/utils/bids.py#L150-L261
* CLI parsing: https://github.com/nipreps/fmriprep/blob/a1974f706ce2250f0535b9f42af395e2fb227a82/fmriprep/cli/parser.py#L179-L190
* Passing filters to `collect_data()`: https://github.com/nipreps/fmriprep/blob/a1974f706ce2250f0535b9f42af395e2fb227a82/fmriprep/workflows/base.py#L156-L162

Would you like to have a go at implementing this?

**Comment by bcmcpher** [comment]
I'd be willing to make an attempt at this. I've figured out the pybids filter parsing in another project recently and having this functionality in MRIQC would be helpful in our group as well. @Remi-Gau helped me take a pass over the process and I have an idea where I'd start.

In addition to the other links, I did find this [function](https://github.com/nipreps/mriqc/blob/562e336b5065c0f3f88fa39fd0f9f50ccaf4e618/mriqc/cli/parser.py#L92-L96) that appears to be unused to load a filter dictionary. Was there a beginning of integrating a filter file that was just never finished?
